### PR TITLE
design: added ceph dashboard sso design doc

### PR DESCRIPTION
**Pull request by gauravsitlani**
Signed-off-by: gauravsitlani <gauravsitlani@riseup.net>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://docs.koor.tech/docs/latest/Contributing/development-flow/#commit-structure)).
- [x] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://docs.koor.tech/docs/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/koor-tech/koor/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: design/ceph/ceph-dashboard-sso.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+       name: my-secret
+   spPrivateKey:
+     key: tls.key
+     secret:
+       name: my-secret
+```
+
+Use https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#secretreference-v1-core or something similar to target a secret in the same namespace for the spCert.secret and spPrivateKey.secret
+
+**Note:P**
```
Comment (galexrt) [inline design/ceph/ceph-dashboard-sso.md:66]:
```suggestion
**Note:**
```

File: design/ceph/ceph-dashboard-sso.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+The issuer value of SAML requests will follow this pattern: `<ceph_dashboard_base_url>/auth/saml2/metadata`
+
+## References
+https://docs.ceph.com/en/latest/mgr/dashboard/#enabling-single-sign-on-sso
+https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/
```
Comment (galexrt) [inline design/ceph/ceph-dashboard-sso.md:71]:
```suggestion
## References

* https://docs.ceph.com/en/latest/mgr/dashboard/#enabling-single-sign-on-sso
* https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/
```

File: design/ceph/ceph-dashboard-sso.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    # urlPrefix: /ceph-dashboard
+    # serve the dashboard at the given port.
+    # port: 8443
+    # serve the dashboard using SSL
+    ssl: true
+           # configure sso for the dashboard access
+    sso:
+               enabled: false # by default it’ll be false
+               autoCreateUsers: true # by default true
+       baseUrl: <url_value>
```
Comment (galexrt) [inline design/ceph/ceph-dashboard-sso.md:49]: Indentation here seems broken

File: design/ceph/ceph-dashboard-sso.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    # serve the dashboard using SSL
+    ssl: true
+           # configure sso for the dashboard access
+    sso:
+      enabled: false # by default it’ll be false
+      autoCreateUsers: true # by default true
+      baseUrl: <url_value>
+      idpMetadataUrl: <url_value>
+      idpAttributes:
+        username: “username”
```
Comment (galexrt) [inline design/ceph/ceph-dashboard-sso.md:52]:
```suggestion
        username: "username"
```

File: design/ceph/ceph-dashboard-sso.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      entityIDd:
```
Comment (galexrt) [inline design/ceph/ceph-dashboard-sso.md:53]:
```suggestion
      #entityIDd: "" # will default to ...
```

File: design/ceph/ceph-dashboard-sso.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+Use https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#secretreference-v1-core or something similar to target a secret in the same namespace for the spCert.secret and spPrivateKey.secret
+
+**Note:**
+The issuer value of SAML requests will follow this pattern: `<ceph_dashboard_base_url>/auth/saml2/metadata`
+
+## References
+
+https://docs.ceph.com/en/latest/mgr/dashboard/#enabling-single-sign-on-sso
+https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/
```
Comment (galexrt) [inline design/ceph/ceph-dashboard-sso.md:72]:
```suggestion
* https://docs.ceph.com/en/latest/mgr/dashboard/#enabling-single-sign-on-sso
* https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/
```

File: design/ceph/ceph-dashboard-sso.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+[...]
+    dashboard:
+    enabled: true
+    # serve the dashboard under a subpath (useful when you are accessing the dashboard via a reverse proxy)
+    # urlPrefix: /ceph-dashboard
+    # serve the dashboard at the given port.
+    # port: 8443
+    # serve the dashboard using SSL
+    ssl: true
+           # configure sso for the dashboard access
```
Comment (galexrt) [inline design/ceph/ceph-dashboard-sso.md:45]: Indentation is off here

**Comment by gauravsitlani** [comment]
hey @galexrt i have updated the design doc like we discussed. Please let me know your feedback regarding the same.

File: design/ceph/ceph-dashboard-sso.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    sso:
+      enabled: false # by default it’ll be false
+      baseUrl: <url_value>
+      idpMetadataUrl: <url_value>
+      idpAttributes:
+        username: "username"
+      entityId: "" # will default to ...
+      users: # list of users with an associated role to be created
+        <username>:
+          <role>
```
Comment (galexrt) [inline design/ceph/ceph-dashboard-sso.md:61]: Let's have this be a list of roles so an user can be be assigned multiple.

```suggestion
          - <role>
```

Comment (gauravsitlani) [reply design/ceph/ceph-dashboard-sso.md]: @galexrt updated.

**Review by galexrt — APPROVED** [review]

**Pull request merged**